content/blog/modern-text-features/index.Rmd

```diff
@@ -0,0 +1,370 @@
+More generally, the OpenType font format describes a long range of features, many optional, that defines specific glyph substitutions (both one-to-one and many-to-one) or position adjustments that can be turned on or off and will affect the look of the final rendered text.
+Some of these features are turned on automatically for specific scripts (e.g. required ligatures for Arabic), while others are left for the user to turn on at their discretion (e.g. tabular numerics).
+As part of the work to add support for non-Latin scripts the infrastructure to support all OpenType features was build.
+This, of course, requires that the font in use supports the requested feature.
+
+Some fonts uses ligatures as a main part of their appeal, and these will now work as advertised with ragg:
+
+```{r, eval=FALSE}
+code <- "x <- y != z"
+logo <- "twitter"
+p <- ggplot() + 
+  geom_text(
+    aes(x = 0, y = 2, label = code), 
+    family = "Fira Code"
+  ) + 
+  geom_text(
+    aes(x = 0, y = 1, label = logo), 
+    family = "Font Awesome 5 brands"
+  ) + 
+  expand_limits(y = c(0, 3))
+
+preview_devices(p, "def_features")
+```
+
+```{r, echo=FALSE, results='asis'}
+get_results("def_features")
+```
+
+But what about non-default features?
+The capabilities of the graphic engine in R presents a problem here.
+There is very little information that the user is able to sent along with the text to be plotted, apart from location and font (**bold** and *italic* on/off is the extend of it).
+So, having a device with support for advanced OpenType features in and off itself is nearly useless as there is no way to specify in your plot code that you want to turn a feature on or off.
+
+In order to get around this without dropping support for the standard ways one puts text in plots with R, systemfonts now allows you to register font features along with a font under a different name.
+The font registration mechanism was previously mostly used for giving access to fonts that were not installed on the system (but e.g. provided by a package), but its use has now expanded and a `register_variant()` function has been added to quickly create a new version of an existing font:
+
+```{r}
+library(systemfonts)
+register_variant(
+  name = "Montserrat Extreme", 
+  family = "Montserrat", 
+  weight = "semibold",
+  features = font_feature(ligatures = "discretionary", letters = "stylistic")
+)
+```
+
+The code above creates a new font based on Montserrat using a light weight and turning on standard ligatures and stylistic letter substitution.
+Now, in your text plotting code all you have to do is specify `"Montserrat Extreme"` as the font family and the features and weights will be used.
+It should be noted that there is no point in comparing with other devices here, since none of the others are build on top of systemfonts and will thus not have accessed to the registered font:
```

Comment:
```suggestion
This only works with ragg, because none of the other devices are build on top of systemfonts, so don't know how to access to the registered font:
```